core/src/main/java/org/frankframework/pipes/ForPipe.java

```diff
@@ -0,0 +1,157 @@
+/*
+   Copyright 2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package org.frankframework.pipes;
+
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.core.ParameterException;
+import org.frankframework.core.PipeForward;
+import org.frankframework.core.PipeLineSession;
+import org.frankframework.core.PipeRunException;
+import org.frankframework.core.PipeRunResult;
+import org.frankframework.doc.ElementType;
+import org.frankframework.doc.ElementType.ElementTypes;
+import org.frankframework.parameters.ParameterValue;
+import org.frankframework.parameters.ParameterValueList;
+import org.frankframework.stream.Message;
+
+/**
+ * ForPipe is a wrapper to use another pipe a fixed number of times. This can be accomplished by something like:
+ *
+ * <pre>{@code
+ * 		<ForPipe name="forPipe" startAt="0" stopAt="10">
+ * 		 	<Forward name="stop" path="EXIT" />
+ * 		 	<Forward name="continue" path="echoPipe"/>
+ * 		</ForPipe>
+ *
+ * 		<EchoPipe name="echoPipe" getInputFromSessionKey="forPipe.iteration">
+ * 		 	<Forward name="success" path="forPipe"/>
+ * 		</EchoPipe>
+ * }</pre>
+ * <p>
+ * This should call the echoPipe for i=0 until i=10.
+ * <p>
+ *
+ * @author evandongen
+ * @ff.info After completing the for loop, the sessionKey containing the for loop iteration state, will be removed.
+ * @ff.info The default format for the session key is "pipeName-iteration"
+ */
+@ElementType(ElementTypes.ITERATOR)
+public class ForPipe extends FixedForwardPipe {
+
+	static final String STOP_FORWARD_NAME = "stop";
+	static final String CONTINUE_FORWARD_NAME = "continue";
+	static final String STOP_AT_PARAMETER_VALUE = "stopAt";
+	private static final String INCREMENT_SESSION_KEY_SUFFIX = "iteration";
+	private @Getter int startAt = 0;
+	private @Getter Integer stopAt = null;
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+
+		// Mandatory stopAt attribute / parameter
+		if (stopAt == null && !getParameterList().hasParameter(STOP_AT_PARAMETER_VALUE)) {
+			throw new ConfigurationException("Value for 'stopAt' is mandatory to break out of the for loop pipe");
+		}
```

Comment:
Wat zou moeten gebeuren als zowel parameter als attribuut geconfigureerd zijn?
Foutmelding, of wordt attribuut-waarde een default in geval de parameter geen value heeft?

Beide opties zijn nu niet afgedekt.

Reply:
attribuut gaat voor parameter. Ik zal dit nog in de javadoc toevoegen

Reply:
En misschien een fout gooien als ze allebei gezet zijn?

Reply:
Goed punt, de fout gooien. Daarmee is de javadoc overbodig. Ook gelijk in een unit test verwerkt.